libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/EdgeToEdgeUtil.kt

```diff
@@ -0,0 +1,36 @@
+package com.salesforce.androidsdk.ui
+
+import android.graphics.drawable.LayerDrawable
+import android.os.Build.VERSION_CODES.UPSIDE_DOWN_CAKE
+import android.view.View
+import androidx.activity.enableEdgeToEdge
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.updatePadding
+import com.salesforce.androidsdk.R.drawable.sf__fix_status_bar
+import com.salesforce.androidsdk.R.id.sf__status_bar_background
+
+// TODO: Remove this in 13.0 after rewriting screens in compose.
+internal fun AppCompatActivity.fixEdgeToEdge(view: View) {
+    if (application.applicationInfo.targetSdkVersion > UPSIDE_DOWN_CAKE) {
+        enableEdgeToEdge()
+        ViewCompat.setOnApplyWindowInsetsListener(view) { listenerView, windowInsets ->
+            val insets = windowInsets.getInsets(
+                WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout()
+            )
+
+            // Fix screens being drawn behind status and navigation bars
+            listenerView.updatePadding(insets.left, insets.top, insets.right, insets.bottom)
+
+            // Fix transparent status bar not matching action bar
+            val background = ResourcesCompat.getDrawable(resources, sf__fix_status_bar, null)
+            val statusBarFiller = (background as LayerDrawable).findDrawableByLayerId(sf__status_bar_background)
+            statusBarFiller.setBounds(0, 0, insets.right, insets.bottom)
+            view.setBackgroundResource(sf__fix_status_bar)
```

Comment:
I think `setBackgroundResource` does a `getDrawable` followed by `setBackground` so we could just do:
`view.setBackground(background)`

Also we might only want to set the background once:
```
if (view.background != background) {
    view.setBackground(background)
}
```

Reply:
I tried the if but every time the breakpoint was hit `view.background` did not equal `background` and I did not seeing it being called multiple times.   I am updating this code though some of it is actually unnecessary.